batch_job/src/batch_job_mesos.c

```diff
@@ -0,0 +1,379 @@
+				string_combine(path_buf, "/");
+				string_combine(path_buf, mt->task_input_files[i]);
+				mt->task_input_files[i] = path_buf;
+			}
+		}	
+	} else {
+		mt->task_input_files = NULL;
+		mt->num_input_files = 0;
+	}
+
+	if (extra_output_files != NULL) {
+		mt->task_output_files = build_str_lst_from_str(&(mt->num_output_files), extra_output_files);
+	} else {
+		mt->task_output_files = NULL;
+		mt->num_output_files = 0;
+	}
+
+	return mt;
+}
+
+void destroy_mesos_task(mesos_task *mt) 
+{
+	free(mt->task_cmd);
+	int i = 0;
+	for (i = 0; i < mt->num_input_files; i++) {
+		free(mt->task_input_files[i]);
+	}
+	free(mt->task_input_files);	
+	int j = 0;
+	for (j = 0; j < mt->num_output_files; j++) {
+		free(mt->task_output_files[j]);
+	}
+	free(mt->task_output_files);
+	free(mt);
+}
+
+static int is_in_array(int a, int *int_array, int size) 
+{
+	int i = 0;
+	for(i = 0; i < size; i++) {
+		if(int_array[i] == a) {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+// check is file is exist
+int is_file_exist(const char *path) 
+{
+	return access(path, F_OK) != -1;
+}
+
+// check the is file is modified since @old_time
+int is_file_modified(const char *path) {
+	struct stat file_stat;
+	int err = stat(path, &file_stat);
+	if (err != 0) {
+		exit(errno);
+	}
+	if (file_stat.st_mtime > old_time) {
+		old_time = file_stat.st_mtime;
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
+static batch_job_id_t batch_job_mesos_submit (struct batch_queue *q, const char *cmd, const char *extra_input_files, const char *extra_output_files, struct jx *envlist, const struct rmsummary *resources )
+{
+	int task_id = ++counter;
+
+	debug(D_BATCH, "task %d is ready", task_id);
+	struct batch_job_info *info = malloc(sizeof(*info));
+	memset(info, 0, sizeof(*info));
+	info->submitted = time(0);
+	info->started = time(0);
+	itable_insert(q->job_table, task_id, info);
+
+	FILE *fp_1;
+
+	if(access(FILE_TASK_INFO, F_OK) != -1) {
+		fp_1 = fopen(FILE_TASK_INFO, "a+");
+	} else {
+		fp_1 = fopen(FILE_TASK_INFO, "w+");
+	}
+	
+	mesos_task *mt = create_mesos_task(task_id, cmd, extra_input_files, extra_output_files);
+
+	fprintf(fp_1, "%d,%s,", mt->task_id, mt->task_cmd);
+
+	// Get the absolut path of each input file
+
+	if (extra_input_files != NULL) {
+
+		int j = 0;
+		for (j = 0; j < (mt->num_input_files-1); j++) {
+			fprintf(fp_1, "%s ", mt->task_input_files[j]);
+		}
+		fprintf(fp_1, "%s,", mt->task_input_files[mt->num_input_files-1]);
+
+	} else {
+		fputs(",", fp_1);
+	}
+    
+	if (extra_output_files != NULL) {
+		int j = 0;
+		for (j = 0; j < (mt->num_output_files-1); j++) {
+			fprintf(fp_1, "%s ", mt->task_output_files[j]);
+		}
+		fprintf(fp_1, "%s,", mt->task_output_files[mt->num_output_files-1]);
+	} else {
+		fputs(",", fp_1);
+	}
+	fputs("submitted\n", fp_1);
+
+	destroy_mesos_task(mt);
+	fclose(fp_1);
+
+	return task_id;
+}
+
+static batch_job_id_t batch_job_mesos_wait (struct batch_queue * q, struct batch_job_info * info_out, time_t stoptime)
+{
+
```

Comment:
Expand this into a more detailed comment (at the top of the file) that explains the relationship between batch_job_mesos and the script.